Shared/Realm.Sync.Shared/Handles/SessionHandle.cs

```diff
@@ -31,20 +32,90 @@ private static class NativeMethods
+            public static extern IntPtr get_user(SessionHandle session);
+
+            [DllImport(InteropConfig.DLL_NAME, EntryPoint = "realm_syncsession_get_uri", CallingConvention = CallingConvention.Cdecl)]
+            public static extern IntPtr get_uri(SessionHandle session, IntPtr buffer, IntPtr buffer_length, out NativeException ex);
+
+            [DllImport(InteropConfig.DLL_NAME, EntryPoint = "realm_syncsession_get_state", CallingConvention = CallingConvention.Cdecl)]
+            public static extern SessionState get_state(SessionHandle session, out NativeException ex);
+
+            [DllImport(InteropConfig.DLL_NAME, EntryPoint = "realm_syncsession_get_from_realm", CallingConvention = CallingConvention.Cdecl)]
+            public static extern IntPtr get_from_realm(SharedRealmHandle realm, out NativeException ex);
+
+            [DllImport(InteropConfig.DLL_NAME, EntryPoint = "realm_syncsession_get_raw_pointer", CallingConvention = CallingConvention.Cdecl)]
+            public static extern IntPtr get_raw_pointer(SessionHandle session);
+
             [DllImport(InteropConfig.DLL_NAME, EntryPoint = "realm_syncsession_destroy", CallingConvention = CallingConvention.Cdecl)]
             public static extern void destroy(IntPtr handle);
         }
 
+        public SyncUserHandle User
+        {
+            get
+            {
+                var handle = new SyncUserHandle();
+                var ptr = NativeMethods.get_user(this);
+                handle.SetHandle(ptr);
+                return handle;
+            }
+        }
+
+        public string ServerUri
+        {
+            get
+            {
+                return MarshalHelpers.GetString((IntPtr buffer, IntPtr length, out bool isNull, out NativeException ex) =>
+                {
+                    isNull = false;
+                    return NativeMethods.get_uri(this, buffer, length, out ex);
+                });
+            }
+        }
+
+        public SessionState State
+        {
+            get
+            {
+                NativeException ex;
+                var state = NativeMethods.get_state(this, out ex);
+                ex.ThrowIfNecessary();
+                return state;
+            }
+        }
+
         public void RefreshAccessToken(string accessToken, string serverPath)
         {
             NativeException ex;
             NativeMethods.refresh_access_token(this, accessToken, (IntPtr)accessToken.Length, serverPath, (IntPtr)serverPath.Length, out ex);
             ex.ThrowIfNecessary();
         }
 
+        public static IntPtr SessionForRealm(SharedRealmHandle realm)
+        {
+            NativeException ex;
+            var ptr = NativeMethods.get_from_realm(realm, out ex);
+            ex.ThrowIfNecessary();
+            return ptr;
+        }
+
         protected override void Unbind()
         {
             NativeMethods.destroy(handle);
         }
+
+        internal class Comparer : IEqualityComparer<SessionHandle>
```

Comment:
This could be public (it's already internal due to being contained in an internal class).